Reindent to 2 spaces. Add support for POST HTTP Body parsing

ajself opened the pull request:
@bellycard/ios Lots of renaming and fixing indents. The meat of this PR is seen at the bottom in Tests/ManagerTests.swift. Please note it is a pending test because of a bug I can't work around at the moment. The functionality does work though and was tested locally. Although PopTop does not have a demo app at the moment I have tested it in another and it performs as expected.

This test came out of a need to refactor resourceArtifactsFromURL to become resourceArtifactsFromRequest located in Source/Manager.swift. This enables PopTop to parse POST HTTPBody data.

Commit:
Reindent to 2 spaces. Add support for POST HTTP Body parsing

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,159 @@
+    /// The `name` is not returned as it is identicical to the `resourceIdentifier` on the `resource` instance
+    let dataToReturn = resource!.data(request, resourceArtifacts: (ids: resourceArtifacts!.ids, query: resourceArtifacts!.query, body: resourceArtifacts!.body))
+
+    let response = NSHTTPURLResponse(URL: request.URL!, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: ["Content-Type": resource!.contentType])!
```
**OUTSIDE THE SCOPE OF THIS PR
A QUESTION FOR THE FUTURE**
how do you feel about ResourceArtifacts containing a status code?

ajself replied:
I'd like to make it configurable with 200 as a default.

Semi-related: I've decided to keep this project progressing at the patch level thus far. Once it we have some semblance of "done", or a more stable API, we can mark it `0.1.0`. At that point maybe we make a roadmap of what is `1.0.0`?

seanwolter replied:
:+1:

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,159 @@
+  /// Return a dictionary with arrays populated with the values of HTTP body data. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromData(data: NSData?) -> QueryArtifacts? {
+    guard let data = data,
+      let componentString = String(data: data, encoding: NSUTF8StringEncoding)
```
I prefer lined up `let`s and `else` on the last line, like:
<img width="311" alt="screen shot 2016-02-04 at 12 48 39 pm" src="https://cloud.githubusercontent.com/assets/68043/12825811/a70d5d62-cb3d-11e5-8acd-7359ea1d4063.png">
Buuuut, what do you think?

ajself replied:
I asked the community at large this same question. I provided a few possibilities in my post but where we ended up in conversation was that Xcode tries to hassle you like a bully and it can be annoying/hard to format things the way you want to.  Here's a snippet of their response:

![screen shot 2016-02-04 at 3 07 52 pm](https://cloud.githubusercontent.com/assets/1699437/12829728/3b8261b4-cb51-11e5-9ae0-e860a1b615e2.png)

But in my opinion I like the style that you posted. What do you think?

seanwolter replied:
Does Xcode auto-indent the third option? I suppose we're hamstrung by Xcode's preferences.

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,159 @@
-        var queryDict = QueryArtifacts()
+  // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
+  /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromString(string: String?) -> QueryArtifacts? {
```
"any func that can be private should be made private"
How do you feel about that guideline?

ajself replied:
Preach :raised_hands: 

Will fix

seanwolter commented:
Can you use `ContiguousArray` since the project is Swift only will never ever need to work with Objective-C?
https://github.com/apple/swift/blob/master/docs/OptimizationTips.rst#advice-use-contiguousarray-with-reference-types-when-nsarray-bridging-is-unnecessary

seanwolter commented:
I read that one thing on optimization and I'm full of beans! I'm jazzed on Swift!!

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,159 @@
+    var queryDict = QueryArtifacts()
 
-            guard parts.count > 1 else { continue }
+    for kVString in string.componentsSeparatedByString("&") {
```
what's `kVString`? are you prefixin'?

seanwolter replied:
The old cocoa in my bones wants variables to be full sentences :neckbeard:

ajself replied:
Hah yeah thats a pretty sloppy name. How about expanding it to "keyValueString". Struggling to find a semantic name to describe the returned array.

seanwolter replied:
:+1:

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,159 @@
-            let key = parts.first!.stringByRemovingPercentEncoding!
-            let value = parts.last!.stringByRemovingPercentEncoding!
-            var values = queryDict[key] ?? [String]()
+      guard parts.count > 1 else { continue }
```
I don't know if this use of guard is brilliant or insane
for me `if parts.count > 1 { ... }` seems easier to visually parse, but that may be because guard is new to me

ajself replied:
Originally, I liked it but rethinking it now this it is too much of a mind bender to understand quickly. I'll opt to switch it to `if parts.count < 2 { continue }` instead which is just as effective and easier to understand.

seanwolter commented on Tests/ManagerTests.swift:
```diff
@@ -155,60 +156,108 @@ class ManagerTests: XCTestCase {
         XCTAssertNil(resourceArtifacts)
     }
+
+    // Skip this test for now due to a bug where HTTPBody is set to nil when PopTop receives the NSURLRequest
```
Is this test disabled in the scheme's Test settings?

seanwolter replied:
Looks like it

ajself replied:
I have a habit of prefixing `x` in front of tests names to indicate a skip, or pending test. This is a carry over from other unit test frameworks. Xcode ignores this test appropriately but do you have another suggestion I should follow instead?

seanwolter replied:
As I understand it, the options are
- disable in Xcode scheme settings
- prefix with `x`
- comment the whole test

I like commenting the entire test because it's more obvious which test is being ignored.

Commit:
Use contiguous array for optimizations and minor updates

ajself commented:
@bellycard/ios take one more look?

seanwolter commented:
LGTM

seanwolter commented on Source/Manager.swift:
```diff
@@ -9,144 +9,158 @@
+      var pathComponents = url.pathComponents else { return nil }
+
+    var name: String?
+    var ids = ContiguousArray<Int>?()
```
makes this `var ids = IDArtifacts?()`?

Commit:
Change back to array for clarity

The arrays should be small in size so making them ContiguousArrays could be an overoptimization
